File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    
    dict_participant = {}
    
    for item in participant:
        dict_participant[item] = dict_participant.get(item, 0) + 1
        
    for item in completion:
        dict_participant[item] = dict_participant[item] - 1
        if not dict_participant[item]:
            del dict_participant[item]
    
    for key in dict_participant.keys():
        answer = key
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string, int> umap;
    
    for (int i = 0; i < participant.size(); i++) {
        if (umap.count(participant[i]) > 0) {
            umap[participant[i]]++;
        } else {
            umap[participant[i]] = 1;
        }
    }
    
    for (int j = 0; j < completion.size(); j++) {
        if (umap[completion[j]] > 0) {
            umap[completion[j]]--;
        }
    }
    
    for (const auto& [k, v] : umap) {
        if (v == 1) {
            answer = k;
            break;
        }
    }
    
    return answer;
}